mbed/src/iot_network.cpp

```diff
@@ -211,17 +215,20 @@ static size_t network_recv(IotNetworkConnection_t pConnection,
     if (res < 0) {
         tr_error("failed to recv data with %d", res);
         return 0;
+    } else if (res == 0) {
+        tr_warning("peer closed socket");
```

Comment:
```suggestion
        tr_debug("peer closed socket");
```
Maybe it's a normal event instead of a warning

Reply:
It is a normal event but one that causes disconnection issues in mbedtls and MQTT. As a debug message it will go unnoticed. I think that the `mbedtls_ssl_close_notify` is only seen when the peer closed the socket first so I'd rather leave it as a warning until the mbedtls issue is resolved.